File: leet_3046_split_array.go
func isPossibleToSplit(nums []int) bool {

    f := make([]int, 101) // `100` incl.
    for _, n := range nums {
        f[n] += 1
        if f[n] == 3 {
            return false
        }
    }
    return true
}

File: leet_3046_split_array.cc
class Solution {
public:
    bool isPossibleToSplit(vector<int>& nums)
    {
        vector<int> f(101, 0);
        for (int n : nums)
        {
            if (f[n] == 2)
                return false;
            f[n]++;
        }
        return true;
    }
};
